plugins/kotlin/j2k/shared/src/org/jetbrains/kotlin/nj2k/JavaToJKTreeBuilder.kt

```diff
@@ -548,12 +554,19 @@ class JavaToJKTreeBuilder(
                 it.withFormattingFrom(this)
             }
 
-        private fun PsiArrayAccessExpression.toJK(): JKExpression =
-            arrayExpression.toJK()
+        private fun PsiArrayAccessExpression.toJK(): JKQualifiedExpression {
+            if (indexExpression?.text != null) {
```

Comment:
Is this condition important? If an empty index expression is an error in Java, we may ignore it and convert it into a similarly erroneous empty index expression in Kotlin